### PR TITLE
Monfaction attitude hate

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
SUMMARY: Features "Monfaction attitude hate"

#### Purpose of change
Long term faction interactions require attitudes that include where some monfactions hate each other above and beyond the PC. I see this sometimes happening between Mycus and Blob but oftentimes they abandon their feud to pursue the PC.

#### Describe the solution
Currently I have just added the enum, string and attitude_map for "hate".  I still need to understand monmove.cpp and possibly monster.cpp to make this actually work but I wanted to save my progress and also possibly see if anyone comments that nothing I've done so far will actually contribute to my goal.

#### Describe alternatives you've considered
Waiting or bribing someone else to do this.

#### Testing
Compiles and accomplishes goal of making monsters with the appropriate faction attitude attack their hated targets over PCs.  

#### Additional context
First attempts at doing anything in C++.

### CODE REVIEW

**Comment by Maleclypse** [comment]
I think `"default_faction": "fungus"` must be hardcoded because they are not in the monsterfaction.json.

**Comment by Maleclypse** [comment]
Ideally I believe if this works it would allow the JSON to look as follows.
```  {
    "type": "MONSTER_FACTION",
    "name": "PrepNet",
    "friendly": [ "Prepnet_Phyle", "human", "PrepNet", "player" ],
    "neutral": "herbivore",
    "by_mood": [ "insect", "small_animal" ],
    "hate": [ "fungus" ]
  }```

**Comment by Maleclypse** [comment]
Or in vanilla it would look like 
```  {
    "type": "MONSTER_FACTION",
    "name": "zombie",
    "friendly": [ "blob", "cult", "zombie_aquatic" ],
    "neutral": [ "small_animal", "insect", "fish" ],
    "by_mood": "bee",
    "hate": "fungus"
  },

File: src/monattack.cpp [hunk 1]
```diff
@@ -4346,8 +4346,9 @@ bool mattack::parrot( monster *z )
 bool mattack::parrot_at_danger( monster *parrot )
 {
     for( monster &monster : g->all_monsters() ) {
-        if( one_in( 20 ) && monster.anger > 0 &&
-            monster.faction->attitude( parrot->faction ) == mf_attitude::MFA_BY_MOOD &&
+        if( one_in( 20 ) && ( monster.faction->attitude( parrot->faction ) == mf_attitude::MFA_HATE ||
+                              ( monster.anger > 0 &&
+                                monster.faction->attitude( parrot->faction ) ) ) == mf_attitude::MFA_BY_MOOD &&
```
Comment (anothersimulacrum) [inline src/monattack.cpp]:
```suggestion
                                monster.faction->attitude( parrot->faction ) == mf_attitude::MFA_BY_MOOD ) ) &&
```

**Pull request merged**